server/game/core/gameSystem/CardTargetSystem.ts

```diff
@@ -259,6 +259,11 @@ export abstract class CardTargetSystem<TContext extends AbilityContext = Ability
             // TODO TOKEN UNITS: the timing for this is wrong, and it needs to not emit a second 'onLeavesPlay' event
             context.game.actions.defeat({ target: card }).resolve(null, context);
         } else {
+            // Attached upgrades should be unattached before moved
+            if (card.isUpgrade() && card.isAttached()) {
```

Comment:
Sent a message about this in the discord thread

Reply:
Updated to use the `Contract.assertTrue` check.